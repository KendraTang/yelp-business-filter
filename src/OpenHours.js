import React from 'react'
import PropTypes from 'prop-types'
import _ from 'lodash'

import OpenHour from './OpenHour'

const OpenHours = (hours) => (
  <div>
    {
      _(hours)
        .map('open')
        .flatten()
        .groupBy('day')
        .entries()
        .sortBy('0')
        .map(([day, hour]) =>
          <OpenHour
            key={day}
            day={parseInt(day, 10)}
            hour={_.sortBy(hour, 'start')}
          />
        )
        .value()
    }
  </div>
)

OpenHours.propTypes = PropTypes.array

export default OpenHours
