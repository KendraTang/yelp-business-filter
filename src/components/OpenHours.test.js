import OpenHours from './OpenHours'

test('sort by day and start hour', () => {
  const hours = [{
    open: [
      { is_overnight: true, day: 0, start: '2300', end: '0200' },
      { is_overnight: false, day: 2, start: '1730', end: '2000' },
      { is_overnight: false, day: 0, start: '1730', end: '2000' },
      { is_overnight: false, day: 0, start: '1100', end: '1430' },
      { is_overnight: false, day: 1, start: '1100', end: '1430' },
      { is_overnight: false, day: 2, start: '1100', end: '1430' },
      { is_overnight: false, day: 1, start: '1730', end: '2000' },
    ]
  }]
  expect(OpenHours(hours)).toMatchSnapshot()
})

test('multiple opens', () => {
  const hours = [
    {
      open: [
        { is_overnight: false, day: 0, start: '1100', end: '1430' },
      ]
    },
    {
      open: [
        { is_overnight: false, day: 1, start: '1100', end: '1430' },
      ]
    },
  ]
  expect(OpenHours(hours)).toMatchSnapshot()
})
