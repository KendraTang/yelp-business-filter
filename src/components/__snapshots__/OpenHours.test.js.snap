// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple opens 1`] = `
<div>
  <OpenHour
    day={0}
    hour={
      Array [
        Object {
          "day": 0,
          "end": "1430",
          "is_overnight": false,
          "start": "1100",
        },
      ]
    }
  />
  <OpenHour
    day={1}
    hour={
      Array [
        Object {
          "day": 1,
          "end": "1430",
          "is_overnight": false,
          "start": "1100",
        },
      ]
    }
  />
</div>
`;

exports[`sort by day and start hour 1`] = `
<div>
  <OpenHour
    day={0}
    hour={
      Array [
        Object {
          "day": 0,
          "end": "1430",
          "is_overnight": false,
          "start": "1100",
        },
        Object {
          "day": 0,
          "end": "2000",
          "is_overnight": false,
          "start": "1730",
        },
        Object {
          "day": 0,
          "end": "0200",
          "is_overnight": true,
          "start": "2300",
        },
      ]
    }
  />
  <OpenHour
    day={1}
    hour={
      Array [
        Object {
          "day": 1,
          "end": "1430",
          "is_overnight": false,
          "start": "1100",
        },
        Object {
          "day": 1,
          "end": "2000",
          "is_overnight": false,
          "start": "1730",
        },
      ]
    }
  />
  <OpenHour
    day={2}
    hour={
      Array [
        Object {
          "day": 2,
          "end": "1430",
          "is_overnight": false,
          "start": "1100",
        },
        Object {
          "day": 2,
          "end": "2000",
          "is_overnight": false,
          "start": "1730",
        },
      ]
    }
  />
</div>
`;
